src/mongo/db/write_concern_options.h

```diff
@@ -127,52 +121,92 @@ struct WriteConcernOptions {
      *      - Implicit default majority WC is being used.
      */
     bool isImplicitDefaultWriteConcern() const {
-        return usedDefaultConstructedWC || _provenance.isImplicitDefault();
+        return isDefaultConstructed() || _provenance.isImplicitDefault();
     }
 
-    SyncMode syncMode;
+    /**
+     * True if the default constructed WC ({w:1}) was used.
+     *      - Implicit default WC when value of w is {w:1}.
+     *      - Internal commands set empty WC ({writeConcern: {}}), then default constructed WC (w:1)
+     *        is used.
+     * False otherwise:
+     *      - Implicit default WC when value of w is {w:"majority"}.
+     *      - Cluster-wide WC.
+     *          - with (w) value set, for example ({writeConcern: {w:1}}).
+     *          - without (w) value set, for example ({writeConcern: {j: true}}).
+     *      - Client-supplied WC.
+     *          - with (w) value set, for example ({writeConcern: {w:1}}).
+     *          - without (w) value set, for example ({writeConcern: {j: true}}).
+     */
+    bool isDefaultConstructed() const {
+        return _defaultConstructed;
+    }
+
+    /**
+     * Used only for tracking opWriteConcernCounters metric.
+     * True if the (w) value of the write concern used is not set explicitly by client:
+     *      - Default constructed WC ({w:1})
+     *      - Implicit default majority WC.
+     *      - Cluster-wide WC.
+     *          - with (w) value set, for example ({writeConcern: {w:1}}).
+     *          - without (w) value set, for example ({writeConcern: {j: true}}).
+     *      - Client-supplied WC without (w) value set, for example ({writeConcern: {j: true}}).
+     *      - Internal commands set empty WC ({writeConcern: {}}).
+     */
+    bool hasExplicitWValue() const {
+        return _explicitWValue;
+    };
+
+    SyncMode syncMode() const {
+        return _syncMode;
+    }
 
     // The w parameter for this write concern. The wMode represents the string format and
     // takes precedence over the numeric format wNumNodes.
-    int wNumNodes;
-    std::string wMode;
+    int wNumNodes() const {
+        return _wNumNodes;
+    }
+
+    StringData wMode() const {
+        return _wMode;
+    }
+
+    int wTimeout() const {
+        return _wTimeout;
+    }
+
+    Date_t wDeadline() const {
+        return _wDeadline;
+    }
+
+    // TODO: this is only used in one test
+    // `NodeReturnsWriteConcernFailedWhenAWriteConcernTimesOutBeforeBeingSatisified`
+    void setDeadline_forTest(const Date_t& deadline) {
```

Comment:
This is only used in [one test](https://github.com/mongodb/mongo/blob/b7181e6813fbc771f4225d8161b4bd7302630c5c/src/mongo/db/repl/replication_coordinator_impl_test.cpp#L1288-L1331). I think we should probably outright remove the concept of `wDeadline` since its only used in that test, but also think that should be follow-on work.